js/ui/collection/ui.collection_widget.edit.js

```diff
@@ -579,6 +607,22 @@ var CollectionWidget = BaseCollectionWidget.inherit({
             return;
         }
 
+        if(args.name === "items") {
+            if(this.option("repaintChangesOnly")) {
```

Comment:
Use `&&` operator.
Maybe would be better to extract it? And handle `items` changing like an other options. Something like that:
```
case "items":
  var isItemsUpdated = this.partialRefresh(args.value);
  if(!isItemsUpdated) {
    this.callBase(args);
  }
  break;